### ISSUE TITLE
When an SVG element is being detached, remove its events

### ISSUE THREAD

**Issue by wflemingnz-retired** [ngx-d3timeline/ngx-d3timeline#521]
In `NgxD3timelineComponent`, the setter of svgElement attaches events to the SVG when the SVG element itself is attached to the DOM.

We likely should also detach the events in the case where the SVG element is removed or changed. This can be achieved by:

```
ElementRef<SVGElement> _svgElement;
@ViewChild('svgEl') set svgElement(el: ElementRef<SVGElement>) {
  if(_svgElement){
    this.detachZoom(_svgElement);
  }

  _svgElement = el;
  this.setupZoom(_svgElement);
}
```

**Comment by umarmohammed** [comment]
Could we remove the event handler in onDestroy?

**Comment by wflemingnz-retired** [comment]
We could but as I understand it, it wouldn't be as comprehensive as the method I gave.

Consider if view is set(SVG element A), then unset and set again(SVG element B).

Using onDestroy, events will be detached from B but not A.

**Comment by umarmohammed** [comment]
I'm not sure that its possible to unset without destroying the component

**Comment by wflemingnz-retired** [comment]
If you assign 0,0 to view (or later set view to null), I thought the elements in the ngIf are removed from the DOM - this includes the SVG element. 

If inconclusive we could test it.

**Comment by umarmohammed** [comment]
Good point. I guess once we start using the view selectors, this should prevent the svg from getting removed.

**Comment by wflemingnz-retired** [comment]
I think we might still need the ngIf, as the view has not yet been populated via the ResizeObserver when the app first starts up.
